src/modules/cmdpal/Microsoft.CmdPal.UI/MainWindow.xaml.cs

```diff
@@ -479,18 +506,41 @@ private void HandleSummon(string commandId)
         var isRootHotkey = string.IsNullOrEmpty(commandId);
         PowerToysTelemetry.Log.WriteEvent(new CmdPalHotkeySummoned(isRootHotkey));
 
+        var isVisible = this.Visible;
+        unsafe
+        {
+            // We need to check if our window is cloaked or not. A cloaked window is still
+            // technically visible, because SHOW/HIDE != iconic (minimized) != cloaked
+            // (these are all separate states)
+            long attr = 0;
+            PInvoke.DwmGetWindowAttribute(_hwnd, DWMWINDOWATTRIBUTE.DWMWA_CLOAKED, &attr, sizeof(long));
+            if (attr == 1 /* DWM_CLOAKED_APP */)
+            {
+                isVisible = false;
+            }
+        }
+
         // Note to future us: the wParam will have the index of the hotkey we registered.
         // We can use that in the future to differentiate the hotkeys we've pressed
         // so that we can bind hotkeys to individual commands
-        if (!this.Visible || !isRootHotkey)
+        if (!isVisible || !isRootHotkey)
         {
             Activate();
 
             Summon(commandId);
         }
         else if (isRootHotkey)
         {
-            PInvoke.ShowWindow(_hwnd, SHOW_WINDOW_CMD.SW_HIDE);
+            // If there's a debugger attached...
+            if (System.Diagnostics.Debugger.IsAttached)
+            {
+                // ... then manually hide our window. When debugged, we won't get the cool cloaking,
+                // but that's the price to pay for having the HWND not light-dismiss while we're debugging.
+                PInvoke.ShowWindow(_hwnd, SHOW_WINDOW_CMD.SW_HIDE);
```

Comment:
Who shows it again if there's a debugger attached?

Reply:
`ShowHwnd` actually does always call `SW_SHOW`. That just no-ops if the window is already shown